xbmc/addons/interfaces/Filesystem.cpp

```diff
@@ -426,6 +428,35 @@ char* Interface_Filesystem::translate_special_protocol(void* kodiBase, const cha
   return strdup(CSpecialProtocol::TranslatePath(strSource).c_str());
 }
 
+bool Interface_Filesystem::get_disk_space(
+    void* kodiBase, const char* path, uint64_t* capacity, uint64_t* free, uint64_t* available)
+{
+  using namespace KODI::PLATFORM::FILESYSTEM;
+
+  CAddonDll* addon = static_cast<CAddonDll*>(kodiBase);
+  if (addon == nullptr || path == nullptr || capacity == nullptr || free == nullptr ||
+      available == nullptr)
+  {
+    CLog::Log(
+        LOGERROR,
+        "Interface_Filesystem::{} - invalid data (addon='{}', path='{}, capacity='{}, free='{}, "
+        "available='{})",
+        __FUNCTION__, kodiBase, static_cast<const void*>(path), static_cast<void*>(capacity),
+        static_cast<void*>(free), static_cast<void*>(available));
+    return false;
+  }
+
+  std::error_code ec;
+  auto freeSpace = space(CSpecialProtocol::TranslatePath(path), ec);
```

Comment:
Where is `space()` defined?

Reply:
Is here, also searched a while to find:
- https://github.com/xbmc/xbmc/blob/master/xbmc/platform/Filesystem.h#L25
- https://github.com/xbmc/xbmc/blob/master/xbmc/platform/posix/Filesystem.cpp#L34
- https://github.com/xbmc/xbmc/blob/master/xbmc/platform/win32/Filesystem.cpp#L23